### PR TITLE
[Messenger] Describe the doctrine transport

### PR DESCRIPTION
**Pull request by vincenttouzet**
Add documentation relative to the PR symfony/symfony#29007

TODO

- [ ] Document the max granularity in seconds for delay (https://github.com/doctrine/dbal/issues/2873)

### CODE REVIEW

File: messenger/doctrine-transport.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                        'options' => array('table_name' => 'custom_table_name_for_messages'),
+                    ),
+                ),
+            ),
+        ));
+
+Working with multiple connections
+---------------------------------
+
+If you have multiple Doctrine connections defined you can choose the desired one. If you have a connection named `legacy` the you must use the following DSN : `doctrine://legacy`.
```
Comment (nicolas-grekas) [inline messenger/doctrine-transport.rst]: [...] then you should use the following DSN: [...]

File: messenger/doctrine-transport.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
+.. index::
+   single: Messenger; Use doctrine as a transport
+
+Use Doctrine as transport
+=========================
+
+The Messenger component comes with a Doctrine transport. This transport let doctrine handle the storage of your messages. To use this transport you need to define the transport as the following :
```
Comment (nicolas-grekas) [inline messenger/doctrine-transport.rst]: no space before `:`

Comment (nicolas-grekas) [reply messenger/doctrine-transport.rst]: also the word "transport" is repeated a bit too much I think :)

File: messenger/doctrine-transport.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
+.. index::
+   single: Messenger; Use doctrine as a transport
+
+Use Doctrine as transport
+=========================
+
+The Messenger component comes with a Doctrine transport. This let doctrine handle the storage of your messages. To use it you need to define the transport as the following:
```
Comment (magnusnordlander) [inline messenger/doctrine-transport.rst]: "this let" => "this lets"

**Comment by alexander-schranz** [comment]
Should we create for each transport a own documentation file?

In this case we should also create a documentation file about the amqp implementation.
And make a list in https://symfony.com/doc/current/messenger.html#transports

/cc @sroze

File: messenger/doctrine-transport.rst [hunk 4]
```diff
@@ -0,0 +1,214 @@
+.. index::
+   single: Messenger; Use doctrine as a transport
+
+Use Doctrine as transport
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Use Doctrine as Transport

File: messenger/doctrine-transport.rst [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+=========================
+
+The Messenger component comes with a Doctrine transport. This lets doctrine handle the storage of your messages. To use it you need to define the transport as the following:
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Doctrine or doctrine?

File: messenger/doctrine-transport.rst [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/messenger.yaml
+        framework:
+            messenger:
+                transports:
+                    doctrine: "doctrine://default"
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Please remove the quotes

File: messenger/doctrine-transport.rst [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+
+    .. code-block:: xml
+
+        <!-- config/packages/messenger.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Please use httpS for xsd URL’s

File: messenger/doctrine-transport.rst [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+                http://symfony.com/schema/dic/symfony
+                http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Please use httpS for xsd URL’s

File: messenger/doctrine-transport.rst [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+            <framework:config>
+                <framework:messenger>
+                    <framework:transport name="doctrine" dsn="doctrine://default" />
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Please remove the space before “/>”

File: messenger/doctrine-transport.rst [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+                </framework:messenger>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/messenger.php
+        $container->loadFromExtension('framework', array(
+            'messenger' => array(
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Please use short array syntax for example

File: messenger/doctrine-transport.rst [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+                'transports' => array(
+                    'doctrine' => 'doctrine://default',
+                ),
+            ),
+        ));
+
+The format of the DSN is `doctrine://<connection_name>`. The connection name is the name you used in the doctrine configuration. If you only use one connection then use `default`.
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Please use double ticks for inline code

File: messenger/doctrine-transport.rst [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+
+If you have multiple Doctrine connections defined you can choose the desired one. If you have a connection named `legacy` the you should use the following DSN : `doctrine://legacy`.
+
+Customize table name
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: “Customize Table Name”

File: messenger/doctrine-transport.rst [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+
+        <!-- config/packages/messenger.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony
+                http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Please use httpS for xsd URL’s

File: messenger/doctrine-transport.rst [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+
+    .. code-block:: xml
+
+        <!-- config/packages/messenger.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Please use httpS for xsd URL’s

File: messenger/doctrine-transport.rst [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+                    <framework:transport name="doctrine" dsn="doctrine://default?table_name=custom_table_name_for_messages" />
+                </framework:messenger>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/messenger.php
+        $container->loadFromExtension('framework', array(
+            'messenger' => array(
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: Please use short array syntax for example

File: messenger/doctrine-transport.rst [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+                'transports' => array(
+                    'doctrine' => 'doctrine://default?queue_name=custom_queue',
+                ),
+            ),
+        ));
+
+Available options
+-----------------
+
+Options for the transport can be configure on the DSN or as options of the transport.
```
Comment (OskarStark) [inline messenger/doctrine-transport.rst]: configure -> configured 

But the sentence sounds a bit weird to me 🤔

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: messenger/doctrine-transport.rst (hunk 16 above)
Comment (vincenttouzet) [reply messenger/doctrine-transport.rst]: I re-read this and I found this weird too. The `options` repetition is the problem I think.

Maybe : "The transport can be configured via DSN or as options" ? :thinking:

Comment (OskarStark) [reply messenger/doctrine-transport.rst]: Yes, it sounds better 👍

**Comment by weaverryan** [comment]
Thank you Vincent for this great feature and its docs!

**Pull request closed without merging**